Editorial review: Document customizable <select> elements (#38470)
mdn/content: files/en-us/web/css/_doublecolon_checkmark/index.md

The `::checkmark` selector is useful for example if you want to hide the checkmark, use a custom icon, or adjust the checkmark's rendering position inside `<option>` elements.

## Examples

### Customizing the checkmark

To opt-in to customizable select functionality, the `<select>` element and its picker both need to have an {{cssxref("appearance")}} value of `base-select` set on them:

```css
select,
::checkmark(select) {
  appearance: base-select;
}
```

Assuming [flexbox](/en-US/docs/Web/CSS/CSS_flexible_box_layout) is being used to lay out the `<option>` elements (which it is by default in customizable selects), you could then move the checkmark from the start of the row to the end by setting an {{cssxref("order")}} value on it greater than `0`, and aligning it to the end of the row using an `auto` {{cssxref("margin-left")}} value (see [Alignment and auto margins](/en-US/docs/Web/CSS/CSS_box_alignment/Box_alignment_in_flexbox#alignment_and_auto_margins)).

The value of the {{cssxref("content")}} property could also be set to a different emoji, to set a different icon to display.

```css
option::checkmark {
  order: 1;
  margin-left: auto;

>   content: "☑️";

Reviewer: per previous comment - this should be marked as decorative.  This accessibility consideration should be called out in this article as well, that if the replaced content is not marked as decorative, then it'll be included as part of the option's accessible name - and that is not recommended.

---
title: ::checkmark
slug: Web/CSS/::checkmark
page-type: css-pseudo-element
browser-compat: css.selectors.checkmark
---

{{CSSRef}}

> The **`::checkmark`** [CSS](/en-US/docs/Web/CSS) [pseudo-element](/en-US/docs/Web/CSS/Pseudo-elements) targets the checkmark placed inside the currently-selected {{htmlelement("option")}} element of a [customizable select element](/en-US/docs/Learn_web_development/Extensions/Forms/Customizable_select) to provide a visual indication of which one is selected.

Reviewer 2: Is "targets" the right word? I'd tend to use that as a synonym for selects OR if there is a checkmark by default (which depends on the UA probably). 
Perhaps:

```suggestion
The **`::checkmark`** [CSS](/en-US/docs/Web/CSS) [pseudo-element](/en-US/docs/Web/CSS/Pseudo-elements) styles a checkmark placed inside the currently-selected {{htmlelement("option")}} element of a [customizable select element](/en-US/docs/Learn_web_development/Extensions/Forms/Customizable_select). It might be used to provide (or remove) a visual indication of which option is selected.
```

Author: "Targets" is the right word, IMO. A selector targets one or more things to style on the page. You need the whole rule to actually do the styling.

I've updated the last bit as suggested, except I haven't included "(or remove)" — I think it is enough to imply this, and I don't want to encourage people to get rid of useful UX features any more than necessary.

> 

Reviewer 2: See https://github.com/mdn/content/pull/38470/files#r2011152685 - interactive example is usually present, and this would not be very hard to do because you'd just cut back the existing ones.

Author: see above comment.

## Syntax

```css-nolint
::checkmark {
  /* ... */
}

> ```

Reviewer 2: Are there any restrictions on properties that can be styled? For example `::highlight()` only allows a subset.

Author: In this case, no. I do tend to point out the limitations, in such cases.

> ### Parameters
> 
> None.
> 

Reviewer 2: Do pseudo elements ever have parameters? Not sure, but looking at other ones, they don't.

```suggestion
```

Author: Yes, they do. See `::picker()` and `::part()` as examples.

I don't need to add the "Parameters" section if it doesn't take parameters tho, so I've removed it.

## Description

> The `::checkmark` pseudo-element targets the checkmark placed inside a [customizable select element's](/en-US/docs/Learn_web_development/Extensions/Forms/Customizable_select) currently-selected `<option>`.

Reviewer 2: "targets"? (as above)

Author: See above

> It is only available to target when the originating element has a picker and has base appearance set on it via the {{cssxref("appearance")}} property `base-select` value. Its generated box appears before any boxes generated by the {{cssxref("::before")}} pseudo-element, with the icon specified in default browser stylesheet; you can customize it using the {{cssxref("content")}} property.

Reviewer 2: "with the icon specified in default browser stylesheet" - if it is

Author: It is in Chrome, certainly.

Reviewer 2: I guess I'm asking for a clear separation between spec and browser separation, and make it clear that you can customize the icon even if the browser did not have one in the default stylesheet. I _think_ this is better - though we might omit that last sentence altogether:

```suggestion
It is only available to target when the originating element has a picker and has base appearance set on it via the {{cssxref("appearance")}} property `base-select` value. Its generated box appears before any boxes generated by the {{cssxref("::before")}} pseudo-element. The icon can be customized using the {{cssxref("content")}} property. If there is a default icon, as there is in Chrome, it is specified in the default browser stylesheet. 
```

Author: Yes, this is better. I've updated it, but I've omitted the last sentence. I'd rather not include this information if I'm not quite sure about it.

The `::checkmark` selector is useful for example if you want to hide the checkmark, use a custom icon, or adjust the checkmark's rendering position inside `<option>` elements.

> [!NOTE]
> The `::checkmark` pseudo-element is not included in the accessibility tree, so any generated {{cssxref("content")}} set on it will not be announced by assistive technologies. You should still make sure that any new icon you set visually makes sense for its intended purpose.

> 

Author: Not as far as I know.

## Examples

### Customizing the checkmark

To opt-in to customizable select functionality, the `<select>` element and its picker both need to have an {{cssxref("appearance")}} value of `base-select` set on them:

```css
select,
::checkmark(select) {
  appearance: base-select;
}
```

> Assuming [flexbox](/en-US/docs/Web/CSS/CSS_flexible_box_layout) is being used to lay out the `<option>` elements (which it is by default in customizable selects), you could then move the checkmark from the start of the row to the end by setting an {{cssxref("order")}} value on it greater than `0`, and aligning it to the end of the row using an `auto` {{cssxref("margin-left")}} value (see [Alignment and auto margins](/en-US/docs/Web/CSS/CSS_box_alignment/Box_alignment_in_flexbox#alignment_and_auto_margins)).

Reviewer 2: > (which it is by default in customizable selects)

Is that true according to spec, or just Chrome?

Reviewer 2: If it doesn't say so in the spec, doesn't that mean this is Chrome-specific information? I guess we might assume this is true for everything that implements this interface? Either way I'd be tempted to add a caveat like:

Assuming [flexbox](/en-US/docs/Web/CSS/CSS_flexible_box_layout) is being used to lay out the `<option>` elements, which it is true in **current implementations** of customizable selects.

Author: Good call; I've used your wording.

> The value of the {{cssxref("content")}} property could also be set to a different emoji, to set a different icon to display.

Reviewer 2: ```suggestion
The value of the {{cssxref("content")}} property could also be set to a different emoji to change the displayed icon.
```
